geo_to_hca/utils/sra_utils.py

```diff
@@ -31,37 +31,74 @@ def get_srp_accession_from_geo(geo_accession: str) -> [str]:
         raise AssertionError(f'{geo_accession} is not a valid GEO accession')
 
     try:
-        default_params = {
-            'db': 'gds',
-            'retmode': 'json'
-        }
+        response_json = call_esearch(geo_accession, db='gds')
+
+        for summary_id in response_json['idlist']:
+            related_study = find_related_object(summary_id, accession_type='SRP')
+            if related_study:
+                return related_study
+
+            # NOTE: this is a bit too complex, requires some cleanup
+            for sample in find_related_samples(summary_id):
+                sample_esearch_result = call_esearch(sample['accession'], db='gds')
+                for sample_id in sample_esearch_result['idlist']:
+                    experiment_accession = find_related_object(sample_id, accession_type='SRX')
+                    if experiment_accession:
+                        log.debug(f'sample {sample["accession"]} is linked to experiment {experiment_accession}')
+                        related_study = find_study_by_experiment_accession(experiment_accession)
+                        if related_study:
+                            return related_study
```

Comment:
it might be better to write 'else' for readable but in general, everything is really good

Reply:
Regarding your comment about the `else`:
The function searches for the study ID, so when it finds it, it returns. If it does not exist, it continues to the next iteration.

Regardless of the comment, this function is too complex as evident by the many levels of indentation (3 `for`s and 2 `if`s).
I will clean it up.

Reply:
I actually find this level of indentation very easy to follow